docs/en/latest/plugins/datadog.md

```diff
@@ -0,0 +1,128 @@
+| batch_max_size   | integer | optional    | 5000         | [1,...] | Max buffer size of each batch                                                            |
+| inactive_timeout | integer | optional    | 5            | [1,...] | Maximum age in seconds when the buffer will be flushed if inactive                       |
+| buffer_duration  | integer | optional    | 60           | [1,...] | Maximum age in seconds of the oldest entry in a batch before the batch must be processed |
+| max_retry_count  | integer | optional    | 1            | [1,...] | Maximum number of retries if one entry fails to reach dogstatsd server                   |
+
+## Metadata
+
+| Name        | Type    | Requirement |     Default        | Valid         | Description                                                            |
+| ----------- | ------  | ----------- |      -------       | -----         | ---------------------------------------------------------------------- |
+| host        | string  | optional    |  "127.0.0.1"       |               | The DogStatsD server host address                                      |
+| port        | integer | optional    |    8125            |               | The DogStatsD server host port                                         |
+| namespace   | string  | optional    |    "apisix"        |               | Prefix for all the custom metrics sent by APISIX agent. Useful for finding entities for metric graph. e.g. (apisix.request.counter)                                        |
+| constant_tags | array | optional    | [ "source:apisix" ] |              | Static tags embedded into generated metrics. Useful for grouping metric over certain signals. |
+
+To know more about how to effectively write tags, please visit [here](https://docs.datadoghq.com/getting_started/tagging/#defining-tags)
+
+## Exported Metrics
+
+Apache APISIX agent, for every request response cycle, export the following metrics to DogStatsD server if the datadog plugin is enabled:
+
+| Metric Name               | StatsD Type   | Description               |
+| -----------               | -----------   | -------                   |
+| Request Counter           | Counter       | No of requests received.   |
+| Request Latency           | Histogram     | Time taken to process the request (in milliseconds). |
+| Upstream latency          | Histogram     | Time taken to proxy the request to the upstream server till a response is received (in milliseconds). |
+| APISIX Latency            | Histogram     | Time taken by APISIX agent to process the request (in milliseconds). |
+| Ingress Size              | Timer         | Request body size in bytes. |
+| Egress Size               | Timer         | Response body size in bytes. |
+
+## How To Enable
+
+First of all, enable the datadog plugin in the `config.yaml`:
+
+```
+# Add this in config.yaml
+plugins:
+  - ... # plugin you need
+  - datadog
+```
```

Comment:
We don't need to do it. The datadog is enabled by default.

Reply:
You are right, I have added the entry in default.yml.
Removed.